File: app/src/main/java/com/example/user/contactlist/data/local/AppDatabase.java
package com.example.user.contactlist.data.local;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

@Database(entities = {ContactEntity.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase instance;

    public abstract ContactDao contactDao();

    public static AppDatabase getAppDatabase(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(),
                    AppDatabase.class,
                    "app-db")
                    .allowMainThreadQueries()
                    .build();
        }
        return instance;
    }
}

File: app/src/main/java/com/example/user/contactlist/data/local/AppDatabase.kt
package com.example.user.contactlist.data.local

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context

import com.example.user.contactlist.data.model.Contact

@Database(entities = [Contact::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun contactDao(): ContactDao

    companion object {

        private var instance: AppDatabase? = null

        fun getAppDatabase(context: Context): AppDatabase {
            if (instance == null) {
                instance = Room.databaseBuilder<AppDatabase>(context.applicationContext,
                        AppDatabase::class.java!!,
                        "app-db")
                        .allowMainThreadQueries()
                        .build()
            }
            return instance as AppDatabase
        }
    }
}
